filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -2622,7 +2622,15 @@ void OpenGLDriver::readPixels(Handle<HwRenderTarget> src,
         PixelBufferDescriptor& p = *pUserBuffer;
         auto& gl = mContext;
         gl.bindBuffer(GL_PIXEL_PACK_BUFFER, pbo);
-        void* vaddr = glMapBufferRange(GL_PIXEL_PACK_BUFFER, 0,  p.size, GL_MAP_READ_BIT);
+        void* vaddr = nullptr;
+        std::unique_ptr<uint8_t> clientBuffer;
+        if constexpr (HAS_MAPBUFFERS) {
+            vaddr = glMapBufferRange(GL_PIXEL_PACK_BUFFER, 0,  p.size, GL_MAP_READ_BIT);
+        } else {
+            clientBuffer.reset(new uint8_t(p.size));
+            glGetBufferSubData(GL_PIXEL_PACK_BUFFER, 0, p.size, clientBuffer.get());
```

Comment:
I guess you can't use an undefined function with `if constexpr`, you'd have to use #ifdef